src/ClickAwayListener.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import events from './utils/events';
+
+const isDescendant = (el, target) => {
+  if (target !== null) {
+    return el === target || isDescendant(el, target.parentNode);
+  }
+  return false;
+};
+
+const clickAwayEvents = ['mousedown', 'touchstart'];
+const bind = (callback) => clickAwayEvents.forEach((event) => events.on(document, event, callback));
+const unbind = (callback) => clickAwayEvents.forEach((event) => events.off(document, event, callback));
+
+export default class ClickAwayListener extends React.Component {
+
+  static propTypes = {
+    children: React.PropTypes.node,
+    onClickAway: React.PropTypes.any,
+    type: React.PropTypes.any,
+  };
+
+  componentDidMount() {
+    if (this.props.onClickAway) {
+      bind(this.handleClickAway);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.onClickAway !== this.props.onClickAway) {
+      unbind(this.handleClickAway);
+      if (this.props.onClickAway) {
+        bind(this.handleClickAway);
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    unbind(this.handleClickAway);
+  }
+
+  handleClickAway = (event) => {
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    const el = ReactDOM.findDOMNode(this);
+
+    if (document.documentElement.contains(event.target) && !isDescendant(el, event.target)) {
+      this.props.onClickAway(event);
+    }
+  };
+
+
+  render() {
+    const {
+      children,
+      onClickAway,
+      type = 'div',
+      ...other,
+    } = this.props;
+
+    return React.createElement(type, other, children);
```

Comment:
How about rendering children directly? `const el = ReactDOM.findDOMNode(this);` would work anyway. This component no longer needs to render anything extra, it will be like a wrapper that finds it's first node and attached the click away to it. I think it's best to remove `type`:

```js
return children;
```

And then render any dom element that we wish to attach it to as the child?

This is not tested, I don't know if it will work or not. just thinking out load :sweat_smile: